### PR TITLE
Support Configuration of Container Runtimes

### LINKED ISSUE #749: Support configuration of ContainerRuntimes
**What would you like to be added**:

Support configuration of ContainerRuntimes .

Gardener supports the configuration of CRI with containerd (possibly in the future other container managers) instead of using docker. See the [GEP](https://github.com/gardener/gardener/blob/master/docs/proposals/10-shoot-additional-container-runtimes.md).
Additionally low-level container runtimes such as [gvisor can be configured](https://github.com/gardener/gardener-extension-runtime-gvisor) if the gardener extension is installed.

Provide a dashboard configuration to define which container manager (atm: docker or containerd) is the default.

Also see the [configuration section in the Shoot example](https://github.com/gardener/gardener/blob/20c4f58c95ebc7d857572c4919e3af2c768549bf/example/90-shoot.yaml#L31).

**Why is this needed**:

Configuring the Container Runtime is an important feature that  should be exposed to the dashboard users.

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:
This PR adds the functionality to configure Container Runtimes for Workers.
Note: This implementation reflects the current state of implementation in Gardener and it is still under discussion if we are going to merge this as is or if we want to wait until the API is finalized.

Screen Shot 2020-08-28 at 13 22 13
Screen Shot 2020-08-28 at 13 22 05

**Which issue(s) this PR fixes**:
Fixes #749 

**Special notes for your reviewer**:

**Release note**:

```noteworthy user
Added support for configuring Container Runtimes for Workers via the Dashboard
```

### CODE REVIEW

**Comment by petersutter** [comment]
whats the status on this PR @grolu / is the API finalized?

**Comment by grolu** [comment]
Last time I checked with @danielfoehrKn it was not. @danielfoehrKn what is the current status?

**Comment by danielfoehrKn** [comment]
I think at this point we can go ahead with the Dashboard changes as there are no current plans to change the API.

**Comment by petersutter** [comment]
thanks @danielfoehrKn

**Comment by grolu** [comment]
@petersutter I'll take care of this. Need to resolve merge conflicts and also adapt to new theming etc.

File: frontend/src/components/ShootWorkers/ContainerRuntime.vue [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  computed: {
+    containerRuntimeItems () {
+      const containerRuntimes = map(this.machineImageCri, 'name')
+      return uniq([...containerRuntimes, DEFAULT_CONTAINER_RUNTIME])
+    },
+    ociRuntimeItems () {
+      if (this.containerRuntime === DEFAULT_CONTAINER_RUNTIME) {
+        return [DEFAULT_OCI_RUNTIME]
+      }
+      const containerRuntime = find(this.machineImageCri, { name: this.containerRuntime })
```
Comment (holgerkoser) [inline frontend/src/components/ShootWorkers/ContainerRuntime.vue:132]:
```suggestion
      const containerRuntime = find(this.machineImageCri, ['name', this.containerRuntime])
```

File: frontend/src/components/ShootWorkers/ContainerRuntime.vue [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+      return undefined
+    },
+    ociHint () {
+      if (this.immutableCri) {
+        return 'OCI runtimes cannot be changed after worker has been created'
+      }
+      return undefined
+    }
+  },
+  methods: {
```
Comment (holgerkoser) [inline frontend/src/components/ShootWorkers/ContainerRuntime.vue:149]:
```suggestion
  methods: {
    ociItemText (item) {
      return !this.immutableCri && item === this.defaultOci ? `default (${item})` : item
    },
    criItemText (item) {
      return !this.immutableCri && item === this.defaultCri ? `default (${item})` : item
    },
```

File: frontend/src/components/ShootWorkers/ContainerRuntime.vue [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+      default: false
+    }
+  },
+  data () {
+    return {
+      validationErrors,
+      valid: undefined,
+      containerRuntime: undefined,
+      ociRuntime: undefined,
+      defaultCir: DEFAULT_CONTAINER_RUNTIME,
```
Comment (holgerkoser) [inline frontend/src/components/ShootWorkers/ContainerRuntime.vue:118]:
```suggestion
      defaultCri: DEFAULT_CONTAINER_RUNTIME,
```

File: frontend/src/components/ShootWorkers/ContainerRuntime.vue [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+      persistent-hint
+    >
+      <template v-slot:item="{ item }">
+        <span v-if="!immutableCri && item===defaultOci">default ({{item}})</span>
+        <span v-else>{{item}}</span>
+      </template>
+      <template v-slot:selection="{ item }">
+        <span v-if="!immutableCri && item===defaultOci">default ({{item}})</span>
+        <span v-else>{{item}}</span>
+      </template>
```
Comment (holgerkoser) [inline frontend/src/components/ShootWorkers/ContainerRuntime.vue:55]:
```suggestion
      <template v-slot:item="{ item }">
        <span>{{ociItemText(item)}}</span>
      </template>
      <template v-slot:selection="{ item }">
        <span>{{ociItemText(item)}}</span>
      </template>
```

File: frontend/src/components/ShootWorkers/ContainerRuntime.vue [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+      persistent-hint
+    >
+      <template v-slot:item="{ item }">
+        <span v-if="!immutableCri && item===defaultCir">default ({{item}})</span>
+        <span v-else>{{item}}</span>
+      </template>
+      <template v-slot:selection="{ item }">
+        <span v-if="!immutableCri && item===defaultCir">default ({{item}})</span>
+        <span v-else>{{item}}</span>
+      </template>
```
Comment (holgerkoser) [inline frontend/src/components/ShootWorkers/ContainerRuntime.vue:30]:
```suggestion
```suggestion
      <template v-slot:item="{ item }">
        <span>{{criItemText(item)}}</span>
      </template>
      <template v-slot:selection="{ item }">
        <span>{{criItemText(item)}}</span>
      </template>
```
```

File: frontend/src/components/ShootWorkers/WorkerInputGeneric.vue [hunk 6]
```diff
@@ -36,6 +36,14 @@ SPDX-License-Identifier: Apache-2.0
         @valid="onMachineImageValid">
         </machine-image>
       </div>
+      <div class="regularInput">
+        <container-runtime
+        :machine-image-cri="machineImageCri"
+        :worker="worker"
+        :immutable-cri="immutableCri"
+        @valid="onContainerRuntimeValid">
```
Comment (holgerkoser) [inline frontend/src/components/ShootWorkers/WorkerInputGeneric.vue:44]:
```suggestion
          :machine-image-cri="machineImageCri"
          :worker="worker"
          :immutable-cri="immutableCri"
          @valid="onContainerRuntimeValid">
```

**Review by holgerkoser — APPROVED** [review]
\lgtm

**Pull request merged**